introduce client-side print component

LukasLohoff opened the pull request:
## Description

<!-- Please give a short description of the changes you propose. Please use prose and try not to be too technical, if possible. Make sure to mention people that you think should know about the PR. -->

Introduces a client-side print component using [@camptocamp/inkmap](https://github.com/camptocamp/inkmap) which prints the map provided by the MapContext.

Currently supports:
- WMS layers
- WMTS layers
- GeoJSON vector layers
- OSM layers

Todos / open issues:
- support wfs layers
- support styles on feature level (if possible)
- fix issues with geostyler-openlayers-parser:
   - if both fill and stroke are defined, stroke will be silently removed
   - support for named colors
- add legend support (on the way)
- pass configuration options for scale bar, north arrow etc. to inkmap

![print2](https://user-images.githubusercontent.com/5795523/173516953-b1fe6ea4-7d6c-4eb3-abf8-58c370949a79.gif)

@terrestris/devs Please review

## Related issues or pull requests

<!-- Please list issues or pull requests that the changes you propose are related to. It does not matter if they are still open and/or unmerged, any link is appreciated. -->

## Pull request type

<!-- Please check the type of change your PR introduces: -->

<!-- Put an x between the square brackets to check an item, like so: [x] -->

- [ ] Bugfix
- [x] Feature
- [ ] Dependency updates
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe)

## Do you introduce a breaking change?

- [ ] Yes
- [x] No

## Checklist

- [x] I understand and agree that the changes in this PR will be licensed under the
  [BSD-2-Clause](https://github.com/terrestris/react-geo/blob/main/LICENSE).
- [x] I have followed the [guidelines for contributing](https://github.com/terrestris/react-geo/blob/main/CONTRIBUTING.md).
- [x] The proposed change fits to the content of the [Code of Conduct](https://github.com/terrestris/react-geo/blob/main/CODE_OF_CONDUCT.md).
- [x] I have added or updated tests and documentation, and the test suite passes (run `npm test` locally).
- [x] I have added a screenshot/screencast to illustrate the visual output of my update.

marcjansen commented:
Eagerly awaiting this! 🤩

LukasLohoff commented:
Now ready for review!

KaiVolland commented on src/Button/PrintButton/PrintButton.tsx:
```diff
@@ -0,0 +1,240 @@
+  const [loading, setLoading] = useState<boolean>(false);
+  const map = useMap();
+
+  const mapOlLayerToInkmap = async (olLayer: OlLayer): Promise<InkmapLayer | null> => {
```
This could go into an Util class to keep the React component cleaner. It might also be useful somewhere else.

LukasLohoff replied:
moved it to ol-util: https://github.com/terrestris/ol-util/pull/705

KaiVolland reviewed (approved):
Very nice!
Would be cool if you could move the printstuff to an Util class or even better to `base-util` or `ol-util` package.

LukasLohoff commented on src/Button/PrintButton/PrintButton.tsx:
```diff
@@ -0,0 +1,240 @@
+    return null;
+  };
+
+  const generatePrintConfig = async (olMap: OlMap): Promise<InkmapPrintSpec | null> => {
```
same here, see comment above this

LukasLohoff commented:
Had to recreate the commits because my branch was broken. 

New changes since last release: 35ff746 and 6ac8b1c
 - moves inkmap mapping to ol-util

dnlkoch commented:
:tada: This PR is included in version 17.2.0 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/@terrestris/react-geo/v/17.2.0)
- [GitHub release](https://github.com/terrestris/react-geo/releases/tag/v17.2.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
